Modules/Test/classes/class.ilTestExportAbstract.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+/**
+ * @author Fabian Helfer <fhelfer@databay.de>
+ */
+abstract class ilTestExportAbstract
+{
+    protected ilObjTest $test_obj;
+    protected ilLanguage $lng;
+    protected string $filterby;
+    protected string $filtertext;
+    protected bool $passedonly;
+
+    public function __construct(
+        ilObjTest $test_obj,
+        string $filterby = '',
+        string $filtertext = '',
+        bool $passedonly = false
+    ) {
+        global $DIC;
+        $this->lng = $DIC->language();
```

Comment:
If all other dependencies are passed in the constructor, we should consider passing the language as well.